flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/KeyValueFileStoreRead.java

```diff
@@ -50,13 +60,18 @@
  */
 public class KeyValueFileStoreRead implements FileStoreRead<KeyValue> {
 
+    private final TableSchema tableSchema;
     private final DataFileReader.Factory dataFileReaderFactory;
     private final Comparator<RowData> keyComparator;
     private final MergeFunction mergeFunction;
+    private final boolean valueCountMode;
+    private final FieldStatsArraySerializer rowStatsConverter;
+
+    @Nullable private int[][] keyProjectedFields;
 
-    private int[][] keyProjectedFields;
+    @Nullable private List<Predicate> keyFilters;
 
-    private List<Predicate> filters;
+    @Nullable private List<Predicate> allFilters;
```

Comment:
Introduce two filter:
```
private List<Predicate> overlappedFilters;
private List<Predicate> nonOverlappedFilters;
```
So in the `createReader`, there is no need to judge `acceptAllFilter`.